src/game.cpp

```diff
@@ -3505,6 +3505,11 @@ shared_ptr_fast<ui_adaptor> game::create_or_get_main_ui_adaptor()
     return ui;
 }
 
+void game::swap_main_ui_adaptor( weak_ptr_fast<ui_adaptor> &ui )
+{
+    main_ui_adaptor.swap( ui );
+}
+
```

Comment:
Can you explain what does this do?

Reply:
I'm not gonna pretend I understand the ui_adaptor system very clearly, but the zoom_out function, for example, does some calling and ends up at
```cpp
void rescale_tileset( int size )
{
    tilecontext->set_draw_scale( size );
    g->mark_main_ui_adaptor_resize();
}
```
which (eventually) calls the main_ui_adaptor's resize callback. I didn't want to pass parameters around everywhere and wanted to minimize code duplication. So I thought I'd just treat the overmap window as the main ui adaptor and give back the terrain window as the main ui adaptor after the overmap is closed. Previously the resize callback for the overmap ui was only called on creation as zoom_out actually called the terrain window's resize callback, for example.

Of course, maybe if this type of swapping were done more often, it'd be cleaner to not have the swapper deal with the previous main_ui_adaptor, but I didn't want to over engineer the system if it's never going to get used again

Reply:
I see, but I'm a bit worried because sometimes `mark_main_ui_adaptor_resize` is not called because of zooming, but because of the change of size of other UI elements, in which case the actual main UI will miss a resize if `mark_main_ui_adaptor_resize` is called when the overmap UI is on.

Reply:
I can't think of a way that that the swapped main ui would ever need to update like that except if it missed the fact that the window was resized, but I tested it a bit with resizing the window with the overmap ui as main and couldn't get anything weird to happen. I could force resize the main ui after switching back from the overmap ui to be extra safe, but if even if there were a problem, it could be solved by just manually zooming in and out.
Again, korg does have an alternate solution that doesn't involve swapping the main ui adaptor if that would be preferred.